influxdb_iox/src/commands/run/querier.rs

```diff
@@ -0,0 +1,75 @@
+//! Implementation of command line option for running the querier
+
+use observability_deps::tracing::*;
+use querier::{
+    handler::QuerierHandlerImpl,
+    server::{grpc::GrpcDelegate, http::HttpDelegate, QuerierServer},
+};
+use std::sync::Arc;
+use thiserror::Error;
+
+use crate::{
+    clap_blocks::{catalog_dsn::CatalogDsnConfig, run_config::RunConfig},
+    influxdb_ioxd::{
+        self,
+        server_type::common_state::{CommonServerState, CommonServerStateError},
+        server_type::querier::QuerierServerType,
+    },
+};
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("Run: {0}")]
+    Run(#[from] influxdb_ioxd::Error),
+
+    #[error("Invalid config: {0}")]
+    InvalidConfig(#[from] CommonServerStateError),
+
+    #[error("Catalog error: {0}")]
+    Catalog(#[from] iox_catalog::interface::Error),
+
+    #[error("Catalog DSN error: {0}")]
+    CatalogDsn(#[from] crate::clap_blocks::catalog_dsn::Error),
+}
+
+#[derive(Debug, clap::Parser)]
+#[clap(
+    name = "run",
+    about = "Runs in querier mode",
+    long_about = "Run the IOx querier server.\n\nThe configuration options below can be \
+    set either with the command line flags or with the specified environment \
+    variable. If there is a file named '.env' in the current working directory, \
+    it is sourced before loading the configuration.
+
+Configuration is loaded from the following sources (highest precedence first):
+        - command line arguments
+        - user set environment variables
+        - .env file contents
+        - pre-configured default values"
+)]
+pub struct Config {
+    #[clap(flatten)]
+    pub(crate) run_config: RunConfig,
+
+    #[clap(flatten)]
+    pub(crate) catalog_dsn: CatalogDsnConfig,
+}
+
+pub async fn command(config: Config) -> Result<(), Error> {
+    let common_state = CommonServerState::from_config(config.run_config.clone())?;
+
+    let catalog = config.catalog_dsn.get_catalog("querier").await?;
+
+    let metric_registry: Arc<metric::Registry> = Default::default();
```

Comment:
I just changed this in #3837 - sorry!

```suggestion
    let metric_registry: Arc<metric::Registry> = Default::default();

    let catalog = config.catalog_dsn.get_catalog("querier", Arc::clone(&metric_registry)).await?;
```